### PR TITLE
Cloud 481

### PR DESCRIPTION
**Pull request by lee-baines**
New simplified PR. Have removed original plan to managed TLS certs with a Kubernetes job.  Now there is a simple shell script and template file and a values flag to switch cert manager off.

### CODE REVIEW

File: bin/generate-tls.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+# Script can be used to either automatically generate a self-signed TLS certificate, or to provide your own TLS cert and key.
```
Comment (dgoldssfo) [inline bin/generate-tls.sh]: Suggestion: This script creates a secret called wildcard.<namespace>.<domain>, which is inherited by the ingress and can be used by all FR components. You can either provide your own TLS cert and key, or you can have the script generate a self-signed TLS certificate.

File: helm/README.md [hunk 2]
```diff
@@ -101,6 +101,13 @@ To enable any of the products to use HTTPS on their external endpoint, set the f
 useTLS: True
 ```
 
+The default behaviour if useTLS = true, is to use certmanager to manage the certificate request/renewal via letsencrypt.  This is enabled by the following flag:
```
Comment (dgoldssfo) [inline helm/README.md]: Sugg: Put this comment above useTLS: true:

The default behavior is to use TLS and have certmanager to manage the certificate request/renewal via 
letsencrypt.

If you want to use TLS but don't want to certmanager to manage the certificate request/renewal, then set
useTLS to true, set useCertManager to false, and run the script ../bin/generate-tls.sh.

File: helm/openidm/templates/tls.yaml [hunk 3]
```diff
@@ -1,20 +1,22 @@
-{{- if .Values.useTLS }}
+# CertManager certificate object created if 'tls.strategy: CertManager'. 
+# This object triggers a certificate request to LetsEncrypt via the CertManager Operator.
```
Comment (dgoldssfo) [inline helm/openidm/templates/tls.yaml]: Is it "LetsEncrypt" or "letsencrypt"? Have a look through your comments and pls make this consistent one way or the other.

File: helm/openidm/values.yaml [hunk 4]
```diff
@@ -33,8 +33,10 @@ image:
 # override Java JVM options.
 javaOpts: "-Xmx1024m -server -XX:+UseG1GC"
 
-# If you set this to true, you must have a TLS secret with the same name as the FQDN.
+# TLS is enabled by default. See TLS section for configuration options
```
Comment (dgoldssfo) [inline helm/openidm/values.yaml]: See my suggestion in the README for replacing this comment.

File: helm/openig/templates/tls.yaml [hunk 5]
```diff
@@ -1,20 +1,22 @@
-{{- if .Values.useTLS }}
+# CertManager certificate object created if 'tls.strategy: CertManager'.
```
Comment (dgoldssfo) [inline helm/openig/templates/tls.yaml]: See my suggestion in the README for replacing this comment.

File: helm/openig/values.yaml [hunk 6]
```diff
@@ -28,20 +28,16 @@ config:
 # Tomcat JVM options.
 catalinaOpts: "-Xmx512m"
   
-# If you set this to true, you must have a TLS secret with the same name as the FQDN
+# TLS is enabled by default. See TLS section for configuration options
```
Comment (dgoldssfo) [inline helm/openig/values.yaml]: See my suggestion in the README for replacing this comment.

File: bin/generate-tls.sh (hunk 1 above)
Comment (lee-baines) [reply bin/generate-tls.sh]: Ah yes, this is slightly confusing.  You can actually use the script to upload your own cert and keys too.  You just need to provide the path to the keys.  I'll reword it so its clearer.  So the script can be used in both use cases

Comment (lee-baines) [reply bin/generate-tls.sh]: I've made it clearer that the script does both

File: helm/README.md (hunk 2 above)
Comment (lee-baines) [reply helm/README.md]: I changed the 2nd section as suggested but as useTLS isn't enabled by default as part of this PR, I can't change the first section quite yet.  I'll be changing this tomorrow once I've coordinated with Pavel as it impacts his tests.

File: helm/openidm/values.yaml (hunk 4 above)
Comment (lee-baines) [reply helm/openidm/values.yaml]: done

File: helm/openig/templates/tls.yaml (hunk 5 above)
Comment (lee-baines) [reply helm/openig/templates/tls.yaml]: done

File: helm/openig/values.yaml (hunk 6 above)
Comment (lee-baines) [reply helm/openig/values.yaml]: done

**Pull request merged**